precompile/contract/interfaces.go

```diff
@@ -47,6 +71,7 @@ type StateDB interface {
 	Exist(common.Address) bool
 
 	AddLog(addr common.Address, topics []common.Hash, data []byte, blockNumber uint64)
+	GetPredicateStorageSlots(address common.Address) ([]byte, bool)
```

Comment:
We turn the access list of `[]common.Hash` into ` []byte` when calling `PrepareAccessList` here: https://github.com/ava-labs/subnet-evm/blob/master/core/state_transition.go#L346.

Precompiles can then use `GetPredicateStorageSlots` to get the predicate bytes during their execution to see what was validated by the predicate. This will be used by the warp and shared memory precompiles to check the verified messages and imported UTXOs respectively.